test/framework/src/main/java/org/elasticsearch/test/rest/VersionBasedNodesFeatures.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test.rest;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.core.Tuple;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+// TODO[lor]: this needs to be replaced with (historical) feature checks once we have figured out a way of exposing them
+// to rest tests (e.g. through an internal API or similar)
+class VersionBasedNodesFeatures {
+    static class VersionBasedNodesFeaturesBuilder {
+        private final List<List<Boolean>> nodesFeaturesBitmap = new ArrayList<>();
+
+        // TODO: replace with FeatureService historical features when they will be available
+        private final List<Tuple<Version, String>> historicalFeatures = List.of(
```

Comment:
`Map<Version, Set<String>>` might be a better representation. Or reverse it, `Map<String, Version>`

Reply:
If I do not need the lookup functionality I prefer to avoid using maps - sure, this is no biggie since we have O(tens) of features like you said, but still I prefer a simple list if I need to just traverse it anyway.